run-tox-env.yml

```diff
@@ -38,8 +38,10 @@ jobs:
       TMPDIR: $(Agent.TempDirectory)
       PIP_NO_WARN_SCRIPT_LOCATION: '0'
       PIP_DISABLE_PIP_VERSION_CHECK: '1'
-      TOXENV: ${{ job.key }}
+      TOXENV: ${{ coalesce(job.tox_env, job.key) }}
```

Comment:
this seems unrelated 🤔

Reply:
It is very much related and required. Current implementation assumes that job name is the tox environment, which is no longer necessarly. Clearly we need a new name for that job, but we do not need to add an additional tox environment for running it. Mainly we need a way to tell template to run "PIP_CONSTRAINT=... tox -e py27" for a job named job27_constraint. That may be the first case, but it should allow us to use the same pattern if we need for other cases.

If you have better ideas, I am open to hear them.

Reply:
We tend to create dedicated tox environments for this. Why does the user need to remember this magical PIP_CONSTRAINT env-var. Shouldn't there be py27-const environment variable? 

That being said thinking this over, I actually would not want us to test with any six higher than 1.9.0. Can't we just enforce the constraint for all tox environments? (this builds on assumption that six latest will not break earlier, which is I think sane).

Reply:
I doubt this approach scales. Today I add one for py27, what if tomorrow I need the same for py36? what if we want to test calling pip with another options. We will soon end-up with a tox.ini that nobody can understand. Still, if you say that we should just create another tox.ini, I would do it. 

In fact if we create a new tox.ini environment, we no longer need to update the template and we can let this open till we are sure we want it.

BTW, check https://github.com/pypa/virtualenv/pull/1607 apparently the approached worked.